Assets/MixedRealityToolkit/UX/Scripts/Tooltip/ToolTipConnector.cs

```diff
@@ -0,0 +1,258 @@
+            PositionAndYRotation,   // Anchor and pivot will follow target like it's parented, but only on Y axis
+            PositionAndRotation,    // Anchor and pivot will follow target like it's parented
+        }
+
+        public enum OrientTypeEnum
+        {
+            OrientToObject,     // Tooltip will maintain anchor-pivot relationship relative to target object
+            OrientToCamera,     // Tooltip will maintain anchor-pivot relationship relative to camera
+        }
+
+        public enum PivotModeEnum
+        {
+            Manual,         // Tooltip pivot will be set manually
+            Automatic,      // Tooltip pivot will be set relative to object/camera based on specified direction and line length
+        }
+
+        public enum PivotDirectionEnum
+        {
+            Manual,         // Direction will be specified manually
+            North,
+            NorthEast,
+            East,
+            SouthEast,
+            South,
+            SouthWest,
+            West,
+            NorthWest,
+            InFront,
+        }
+        
+        public GameObject Target;
+
+        public FollowTypeEnum FollowType = FollowTypeEnum.AnchorOnly;
+        public PivotModeEnum PivotMode = PivotModeEnum.Manual;
+        public PivotDirectionEnum PivotDirection = PivotDirectionEnum.North;
+        public OrientTypeEnum PivotDirectionOrient = OrientTypeEnum.OrientToObject;
+        public Vector3 ManualPivotDirection = Vector3.up;
+        public Vector3 ManualPivotLocalPosition = Vector3.up;
+        [Range(0f, 2f)]
+        public float PivotDistance = 0.25f;
+
+        private void OnEnable()
+        {
+            if (!FindToolTip())
+                return;
+
+            ManualPivotLocalPosition = transform.InverseTransformPoint (toolTip.PivotPosition);
+        }
+
+        private bool FindToolTip()
+        {
+            if (toolTip == null)
+            {
+                toolTip = GetComponent<ToolTip>();
+            }
+            if (toolTip == null)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private void UpdatePosition() {
+
+            if (!FindToolTip())
+                return;
+
+            if (Target == null)
+                return;
+
+            switch (FollowType)
+            {
+                case FollowTypeEnum.AnchorOnly:
+                default:
+                    // Set the position of the anchor to the target's position
+                    // And do nothing else
+                    toolTip.Anchor.transform.position = Target.transform.position;
+                    break;
+
+                case FollowTypeEnum.PositionOnly:
+                    // Move the entire tooltip transform while maintaining the anchor position offset
+                    toolTip.transform.position = Target.transform.position;
+                    switch (PivotMode)
```

Comment:
let's try to put the nested switch statements in their own methods to prevent confusion and reduce nesting.